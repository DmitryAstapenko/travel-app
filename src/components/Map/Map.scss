.leaflet-container {
  height: 100%;
  width: 100%;
}

.map-container {
  width: 100%;
  height: 500px;
  margin: 10px 0;
  position: relative;
}

.fullscreen-btn {
  display: flex;
  justify-content: center;
  align-items: center;
  position: absolute;
  border: 2px solid rgba(163, 163, 163, 0.8);
  background: white;
  width: 32px;
  height: 30px;
  margin: 10px;
  right: 0px;
  top: 0px;
  z-index: 999;
}
.fullscreen-btn-image {
  width: 20px;
  height: 20px;
}
.fullscreen-btn-span {
  background: white;
  color: black;
  font-size: 16px;
}
