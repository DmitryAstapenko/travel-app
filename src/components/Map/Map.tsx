import L, { LatLngExpression } from "leaflet";
import "leaflet/dist/leaflet.css";
import { MapContainer, Marker, TileLayer, Popup, Polygon } from "react-leaflet";
import React from "react";
import { Button } from "react-bootstrap";
import ICountryData from "../../models/country-data";
import "./Map.scss";
import geoData from "../../assets/geoJson/countries";
import markerIcon from "../../assets/images/marker.png";
import fullScreenImage from "../../assets/images/fullscreen.png";

interface IMapProps {
  countryData: ICountryData | undefined;
}

class Map extends React.Component<IMapProps> {
  icon: L.DivIcon = L.icon({
    iconUrl: markerIcon,
    iconSize: [30, 40],
    iconAnchor: [15, 40],
    popupAnchor: [15, 0],
  });

  zoom = 3;

  redOptions = { color: "red" };

  // eslint-disable-next-line
  fullScreen = (e: any): void => {
    if (!document.fullscreenElement) {
      if (e.currentTarget === e.target) {
        e.nativeEvent.path[1].children[1].requestFullscreen();
      } else {
        e.nativeEvent.path[2].children[1].requestFullscreen();
      }
    }
  };

  // eslint-disable-next-line
  render() {
    const { countryData } = this.props;
    const {
      ISOCode,
      capitalLocation: { coordinates },
      // eslint-disable-next-line
    } = countryData!;
    const latitude = coordinates[0];
    const longitude = coordinates[1];
    const copyObj = JSON.parse(JSON.stringify(geoData));
    // eslint-disable-next-line
    const info: any = copyObj;
    // eslint-disable-next-line
    const data: any = info[ISOCode].map((el: any) => {
      el[0].map((item: number[]) => {
        return item.reverse();
      });
      return el;
    });

    // const { latitude, longitude, data } = this.state;
    const position: LatLngExpression = [latitude, longitude];
    return (
      <div className="map-container">
        <Button
          variant="white"
          className="fullscreen-btn"
          onClick={(e) => {
            this.fullScreen(e);
          }}
        >
          <span className="fullscreen-btn-span">â‡²</span>
        </Button>
        <MapContainer
          center={position}
          zoom={this.zoom}
          scrollWheelZoom={false}
          className="leaflet-container"
        >
          <TileLayer
            attribution="&copy; <a href='https://carto.com/'>carto.com</a> contributors"
            url="https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}.png"
          />
          <Marker icon={this.icon} position={[latitude, longitude]}>
            <Popup>
              <span>Capital of country!</span>
            </Popup>
          </Marker>
          <Polygon pathOptions={this.redOptions} positions={data} />
        </MapContainer>
      </div>
    );
  }
}

export default Map;
