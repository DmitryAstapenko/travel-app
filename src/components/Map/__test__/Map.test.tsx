import React from "react";
import Map from "../Map";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

const setUp = (props) => shallow(<Map {...props} />);

describe("should render map component", () => {
  let component;
  beforeEach(() => {
    const countryData = {
      ISOCode: "IT",
      capitalLocation: {
        coordinates: [35, 28],
        type: "point",
      },
    };
    component = setUp({ countryData });
  });
  it("should render map component", () => {
    expect(toJson(component)).toMatchSnapshot();
  });
});
