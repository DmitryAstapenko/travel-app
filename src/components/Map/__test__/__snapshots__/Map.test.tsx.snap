// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render map component should render map component 1`] = `
<div
  className="map-container"
>
  <Button
    active={false}
    className="fullscreen-btn"
    disabled={false}
    onClick={[Function]}
    variant="dark"
  >
    <img
      alt="fullScreen"
      className="fullscreen-btn-image"
      src="fullscreen.png"
    />
  </Button>
  <MapContainer
    center={
      Array [
        35,
        28,
      ]
    }
    className="leaflet-container"
    scrollWheelZoom={false}
    zoom={3}
  >
    <ForwardRef(LeafComponent)
      attribution="Â© <a href='https://carto.com/'>carto.com</a> contributors"
      url="https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}.png"
    />
    <ForwardRef(ContainerComponent)
      icon={
        NewClass {
          "_initHooksCalled": true,
          "options": Object {
            "iconAnchor": Array [
              15,
              40,
            ],
            "iconSize": Array [
              30,
              40,
            ],
            "iconUrl": "marker.png",
            "popupAnchor": Array [
              15,
              0,
            ],
          },
        }
      }
      position={
        Array [
          35,
          28,
        ]
      }
    >
      <ForwardRef(OverlayComponent)>
        <span>
          Capital of country!
        </span>
      </ForwardRef(OverlayComponent)>
    </ForwardRef(ContainerComponent)>
    <ForwardRef(ContainerComponent)
      pathOptions={
        Object {
          "color": "red",
        }
      }
      positions={
        Array [
          Array [
            Array [
              Array [
                38.23115509699147,
                15.520376010813834,
              ],
              Array [
                37.44404551853782,
                15.160242954171736,
              ],
              Array [
                37.1342194687318,
                15.309897902089006,
              ],
              Array [
                36.6199872909954,
                15.09998823411945,
              ],
              Array [
                36.996630967754754,
                14.335228712632016,
              ],
              Array [
                37.1045313583802,
                13.82673261887993,
              ],
              Array [
                37.61294993748382,
                12.431003859108813,
              ],
              Array [
                38.12638113051969,
                12.570943637755136,
              ],
              Array [
                38.03496552179536,
                13.741156447004585,
              ],
              Array [
                38.143873602850505,
                14.76124922044616,
              ],
              Array [
                38.23115509699147,
                15.520376010813834,
              ],
            ],
          ],
          Array [
            Array [
              Array [
                41.20999136002422,
                9.210011834356266,
              ],
              Array [
                40.5000088567661,
                9.809975213264977,
              ],
              Array [
                39.177376410471794,
                9.669518670295673,
              ],
              Array [
                39.240473334300134,
                9.21481774255949,
              ],
              Array [
                38.90661774347848,
                8.80693566247973,
              ],
              Array [
                39.17184703221662,
                8.428302443077115,
              ],
              Array [
                40.378310858718805,
                8.38825320805094,
              ],
              Array [
                40.95000722916379,
                8.15999840661766,
              ],
              Array [
                40.89998444270523,
                8.709990675500109,
              ],
              Array [
                41.20999136002422,
                9.210011834356266,
              ],
            ],
          ],
          Array [
            Array [
              Array [
                46.76755910906988,
                12.376485223040845,
              ],
              Array [
                46.50930613869119,
                13.806475457421556,
              ],
              Array [
                46.016778062517375,
                13.698109978905478,
              ],
              Array [
                45.591015936864665,
                13.937630242578336,
              ],
              Array [
                45.73669179949542,
                13.141606479554298,
              ],
              Array [
                45.381778062514854,
                12.328581170306308,
              ],
              Array [
                44.88537425391908,
                12.383874952858605,
              ],
              Array [
                44.600482082694015,
                12.261453484759159,
              ],
              Array [
                44.091365871754476,
                12.589237094786483,
              ],
              Array [
                43.58772736263791,
                13.526905958722494,
              ],
              Array [
                42.76100779883248,
                14.029820997787027,
              ],
              Array [
                41.955139675456905,
                15.142569614327954,
              ],
              Array [
                41.96131500911574,
                15.926191033601896,
              ],
              Array [
                41.740294908203424,
                16.169897088290412,
              ],
              Array [
                41.5410822617182,
                15.889345737377795,
              ],
              Array [
                41.179605617836586,
                16.785001661860576,
              ],
              Array [
                40.87714345963224,
                17.519168735431208,
              ],
              Array [
                40.35562490494266,
                18.376687452882578,
              ],
              Array [
                40.168866278639825,
                18.480247023195403,
              ],
              Array [
                39.81077444107325,
                18.2933850440281,
              ],
              Array [
                40.2776710068303,
                17.738380161213286,
              ],
              Array [
                40.44223460546385,
                16.869595981522338,
              ],
              Array [
                39.79540070246648,
                16.448743116937322,
              ],
              Array [
                39.42469981542072,
                17.1714896989715,
              ],
              Array [
                38.902871202137305,
                17.052840610429342,
              ],
              Array [
                38.8435724960824,
                16.635088331781844,
              ],
              Array [
                37.98589874933418,
                16.100960727613057,
              ],
              Array [
                37.90884918878703,
                15.684086948314501,
              ],
              Array [
                38.214592800441864,
                15.68796268073632,
              ],
              Array [
                38.750942491199226,
                15.891981235424709,
              ],
              Array [
                38.96454702407769,
                16.109332309644316,
              ],
              Array [
                39.544072374014945,
                15.718813510814641,
              ],
              Array [
                40.04835683853517,
                15.413612501698822,
              ],
              Array [
                40.17294871679093,
                14.998495721098237,
              ],
              Array [
                40.604550279292624,
                14.70326826341477,
              ],
              Array [
                40.78634796809544,
                14.060671827865264,
              ],
              Array [
                41.188287258461656,
                13.627985060285397,
              ],
              Array [
                41.25308950455562,
                12.88808190273042,
              ],
              Array [
                41.70453481705741,
                12.10668257004491,
              ],
              Array [
                42.35542531998968,
                11.191906365614187,
              ],
              Array [
                42.931462510747224,
                10.511947869517797,
              ],
              Array [
                43.920006822274615,
                10.200028924204048,
              ],
              Array [
                44.03627879493132,
                9.702488234097814,
              ],
              Array [
                44.36633616797954,
                8.88894616052687,
              ],
              Array [
                44.23122813575242,
                8.428560825238577,
              ],
              Array [
                43.76714793555524,
                7.850766635783202,
              ],
              Array [
                43.69384491634918,
                7.435184767291844,
              ],
              Array [
                44.12790110938482,
                7.549596388386163,
              ],
              Array [
                44.25476675066139,
                7.007562290076663,
              ],
              Array [
                45.02851797136759,
                6.749955275101712,
              ],
              Array [
                45.333098863295874,
                7.096652459347837,
              ],
              Array [
                45.70857982032868,
                6.802355177445662,
              ],
              Array [
                45.99114655210067,
                6.843592970414562,
              ],
              Array [
                45.77694774025076,
                7.273850945676685,
              ],
              Array [
                45.82449005795928,
                7.755992058959833,
              ],
              Array [
                46.163642483090854,
                8.31662967289438,
              ],
              Array [
                46.00515086525175,
                8.489952426801295,
              ],
              Array [
                46.036931871111165,
                8.966305779667834,
              ],
              Array [
                46.44021474871698,
                9.182881707403112,
              ],
              Array [
                46.31489940040919,
                9.922836541390353,
              ],
              Array [
                46.483571275409844,
                10.363378126678668,
              ],
              Array [
                46.893546250997446,
                10.442701450246602,
              ],
              Array [
                46.7513585475464,
                11.048555942436508,
              ],
              Array [
                46.94157949481274,
                11.164827915093326,
              ],
              Array [
                47.11539317482644,
                12.153088006243081,
              ],
              Array [
                46.76755910906988,
                12.376485223040845,
              ],
            ],
          ],
        ]
      }
    />
  </MapContainer>
</div>
`;
